shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/security/DatasetSecurityService.java

```diff
@@ -1103,5 +1103,60 @@ public boolean hasRightOnExamination(String examinationUID, String rightStr) thr
 		Long id = studyInstanceUIDHandler.extractExaminationId(examinationUID);
 		return hasRightOnExamination(id, rightStr);
     }
-   
+
+	public boolean HasRightOnEveryDatasetOfProcessings(List<Long> processingIds, String rightStr) {
+		boolean hasRight = true;
+
+		for (Long processingId : processingIds) {
+			if (KeycloakUtil.getTokenRoles().contains("ROLE_ADMIN") || processingId == null) {
+				continue;
+			}
+			Iterable<Dataset> datasets = datasetRepository.findDatasetsByProcessingId(processingId);
+
+			for (Dataset dataset : datasets) {
```

Comment:
I wonder, if the below code exists already on the dataset level?

Reply:
Same as above